docs/basics/networkvariable.md

```diff
@@ -504,6 +504,51 @@ In order to create your own `NetworkVariableBase` derived container, you should:
     - `void ReadDelta(FastBufferReader reader, bool keepDirtyDelta)`
 - Depdending upon your custom `NetworkVariableBase` container, you might look at `NetworkVariable<T>` or `NetworkList` to see how those two examples were implemented.
 
+#### NetworkVariableSerialization&lt;T&gt;
+
+When working with known, non-generic types, it is easy and efficient to read and write using `FastBufferReader.ReadValue` and `FastBufferWriter.WriteValue`, respectively, or for integer types, using `BytePacker` and `ByteUnpacker` to add compression. When working with generic types, however, we have to rely on serializers that are generated based on types discovered during a compile-time code generation process. As such, when working with generic types, you have to a little bit of extra work to tell our code generation algorithm which types to generate serializers for. You can do this in two ways:
+
+The first way is to use `GenerateSerializationForTypeAttribute` to generate serialization for a specific hard-coded type:
+```csharp
+[GenerateSerializationForType(typeof(Foo))]
+public class MyNetworkVariableTypeUsingFoo : NetworkVariableBase {}
+```
+
+However, this is rarely useful for generic types, since if you know the type, you can generally use the more efficient route of calling `FastBufferReader`/`FastBufferWriter` methods directly.
+
+The more common and more useful method is to use `GenerateSerializationForGenericParameterAttribute` to generate serialization for a specific generic parameter in your `NetworkVariable` type:
+```csharp
+[GenerateSerializationForGenericParameter(0)]
+public class MyNetworkVariableType<T> : NetworkVariableBase {}
+```
+
+This attribute takes an integer indicating which parameter in the type to generate serialization for. This value is 0-indexed, so the first type is 0, the second is 1, and so on. The attribute can be placed more than once on one class to generate serialization for multiple types:
+
+```csharp
+[GenerateSerializationForGenericParameter(0)]
+[GenerateSerializationForGenericParameter(1)]
+public class MyNetworkVariableType<TFirstType, TSecondType> : NetworkVariableBase {}
+```
+
+This will generate serialization for both `TFirstType` and `TSecondType`.
+
+Either of these attributes will cause the code generation to generate the following methods that you can use:
+
+```csharp
+NetworkVariableSerialization<T>.Write(FastBufferWriter writer, ref T value);
+NetworkVariableSerialization<T>.Read(FastBufferWriter writer, ref T value);
+NetworkVariableSerialization<T>.Duplicate(in T value, ref T duplicatedValue);
+NetworkVariableSerialization<T>.AreEqual(in T a, in T b);
+```
+
+In cases where the type is dynamically allocated (managed types and collections like NativeArray and NativeList), calling `Read` will read the value in-place whenever possible, avoiding further allocations.
```

Comment:
It means we don't create a new object, but re-use the old one and write the new data into it directly.

Reply:
What happens if the value can't be read in-place? Is it not read at all or is it read as `inplace=False` by default? Is there any way the user can tell if the value can't be read in-place?

Reply:
The only case where a value can't be read in-place is if the type is a managed type and the current value is `null`. It can't be read in-place because there is no current "place" to read to, so it allocates a new one. After that, it reads in-place.